documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,252 @@
+              summary: 'Failed Prometheus configuration reload.',
+              description: 'Prometheus %(prometheusName)s has failed to reload its configuration.' % $._config,
+            },
+          },
+          {
+            alert: 'PrometheusNotificationQueueRunningFull',
+            expr: |||
+              (
+                predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
+              >
+                prometheus_notifications_queue_capacity{%(prometheusSelector)s}
+              )
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'Prometheus alert notification queue predicted to run full in less than 30m.',
+              description: 'Alert notification queue of Prometheus %(prometheusName)s is running full.' % $._config,
+            },
+          },
+          {
+            alert: 'PrometheusErrorSendingAlertsToSomeAlertmanagers',
+            expr: |||
+              (
+                rate(prometheus_notifications_errors_total{%(prometheusSelector)s}[5m])
+              /
+                rate(prometheus_notifications_sent_total{%(prometheusSelector)s}[5m])
+              )
+              * 100
+              > 1
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'Prometheus encounters more than 1% errors sending alerts to a specific Alertmanager.',
+              description: '{{ printf "%%.1f" $value }}%% errors while sending alerts from Prometheus %(prometheusName)s to Alertmanager {{$labels.alertmanager}}.' % $._config,
+            },
+          },
+          {
+            alert: 'PrometheusErrorSendingAlertsToAnyAlertmanager',
+            expr: |||
+              min without(alertmanager) (
+                rate(prometheus_notifications_errors_total{%(prometheusSelector)s}[5m])
+              /
+                rate(prometheus_notifications_sent_total{%(prometheusSelector)s}[5m])
+              )
+              * 100
+              > 3
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Prometheus encounters more than 3% errors sending alerts to any Alertmanager.',
+              description: '{{ printf "%%.1f" $value }}%% minimum errors while sending alerts from Prometheus %(prometheusName)s to any Alertmanager.' % $._config,
+            },
+          },
+          {
+            alert: 'PrometheusNotConnectedToAlertmanagers',
+            expr: |||
+              prometheus_notifications_alertmanagers_discovered{%(prometheusSelector)s} < 1
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'warning',
```

Comment:
This could potentially result in not being alerted at all (and also this one is likely to be hard to be sent at all), should this be critical or do we expect users to have dead-mans-switch type alerting setup (if so we should probably at least mention it somewhere)?

Reply:
“Production-ready” meta-monitoring should not be done by a Prometheus server monitoring itself. Thus, this alert will fire if the metamon-prometheus detects other Prometheus servers without any discovered Alertmanagers.

Of course, the metamon-prometheus itself still needs to be able to send alerts. That's where a dead-man-switch like setup enters the game. In general, I think the metamon-prometheus is a good location to test the whole alerting chain with a dead-man-switch like setup. But I believe discussing that aspect of meta-monitoring is out of scope of this mixin example.